airbyte-webapp/src/packages/cloud/views/auth/OAuthLogin/OAuthLogin.tsx

```diff
@@ -0,0 +1,107 @@
+import { useRef, useState } from "react";
+import { FormattedMessage, useIntl } from "react-intl";
+import { useUnmount } from "react-use";
+import { Subscription } from "rxjs";
+
+import { Spinner } from "components";
+
+import { useExperiment } from "hooks/services/Experiment";
+import { useAuthService } from "packages/cloud/services/auth/AuthService";
+
+import githubLogo from "./assets/github-logo.svg";
+import googleLogo from "./assets/google-logo.svg";
+import styles from "./OAuthLogin.module.scss";
+
+const GitHubButton: React.FC<{ onClick: () => void }> = ({ onClick }) => {
+  return (
+    <button className={styles.github} onClick={onClick}>
+      <img src={githubLogo} alt="" />
+      <FormattedMessage id="login.oauth.github" tagName="span" />
+    </button>
+  );
+};
+
+const GoogleButton: React.FC<{ onClick: () => void }> = ({ onClick }) => {
+  return (
+    <button className={styles.google} onClick={onClick}>
+      <img src={googleLogo} alt="" />
+      <FormattedMessage id="login.oauth.google" tagName="span" />
+    </button>
+  );
+};
+
+export const OAuthLogin: React.FC = () => {
+  const { formatMessage } = useIntl();
+  const { loginWithOAuth } = useAuthService();
+  const stateSubscription = useRef<Subscription>();
+  const [errorCode, setErrorCode] = useState<string>();
+  const [isLoading, setLoading] = useState(false);
+
+  const isGitHubLoginEnabled = useExperiment("authPage.oauth.github", false);
+  const isGoogleLoginEnabled = useExperiment("authPage.oauth.google", false);
+
+  const isAnyOauthEnabled = isGitHubLoginEnabled || isGoogleLoginEnabled;
+
+  useUnmount(() => {
+    stateSubscription.current?.unsubscribe();
```

Comment:
I am not sure I follow the question exactly. This will only need to unsubscribe whenever we have a subscription (i.e. we subscribed in the `login()` method below?

Reply:
I think Edmundo is worried when you reassign `current` the previous reference may still persist and need to be unsubscribed?

Reply:
Got you, resolved.